src/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -1594,6 +1609,22 @@ static void AddMSBuildContentForItemOptions(StringBuilder builder)
                 }
             }
 
+            static string GetCodeAnalysisTreatWarningsAsErrorsTargetContents()
+            {
+                return $@"
+  <!--
+    Design-time target to handle 'CodeAnalysisTreatWarningsAsErrors = false' for the CA rule ids implemented in this package.
+    Note that a similar 'WarningsNotAsErrors' property group is present in the generated props file to ensure this functionality on command line builds.
+  -->
+  <Target Name=""_CodeAnalysisTreatWarningsAsErrors"" BeforeTargets=""CoreCompile"" Condition=""'$(DesignTimeBuild)' == 'true' OR '$(BuildingProject)' != 'true'"">
+    <PropertyGroup>
+      <EffectiveCodeAnalysisTreatWarningsAsErrors Condition=""'$(EffectiveCodeAnalysisTreatWarningsAsErrors)' == ''"">$(CodeAnalysisTreatWarningsAsErrors)</EffectiveCodeAnalysisTreatWarningsAsErrors>
+      <WarningsNotAsErrors Condition=""'$(EffectiveCodeAnalysisTreatWarningsAsErrors)' == 'false' and '$(TreatWarningsAsErrors)' == 'true'"">$(WarningsNotAsErrors);$(CodeAnalysisRuleIds)</WarningsNotAsErrors>
+    </PropertyGroup>
+  </Target>
```

Comment:
Isn't this the same logic as in props? I'm not quite sure what is the need to wrap it in a target that runs only for design-time builds.

Reply:
This was added earlier to account for a regression. I am just restoring the old functionality here for `WarningsNotAsErrors`. Either way, I personally don't like any of the current `CodeAnalysisTreatWarningsAsErrors` implementations. I am going to drive implementing this in the core analyzer driver in Roslyn and delete all the code we have in this repo related to  `CodeAnalysisTreatWarningsAsErrors`.